components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/utils/OpenAPICodegenUtils.java

```diff
@@ -702,11 +722,26 @@ private static void validateSingleResourceInterceptors(Operation operation, Stri
      */
     private static void setMgwAPISecurityAndScopes(ExtendedAPI api, OpenAPI openAPI) {
         String[] securitySchemasAndScopes = generateMgwSecuritySchemasAndScopes(openAPI.getSecurity());
+        Map<String, Object> apiDefExtensions = openAPI.getExtensions();
+        if (apiDefExtensions != null && apiDefExtensions.containsKey(OpenAPIConstants.APPLICATION_SECURITY)) {
+            logger.debug(OpenAPIConstants.APPLICATION_SECURITY + " extension found in the API");
+            try {
+                Object appSecurityExtension = apiDefExtensions.get(OpenAPIConstants.APPLICATION_SECURITY);
+                ApplicationSecurity appSecurity =
+                        new ObjectMapper().convertValue(appSecurityExtension, ApplicationSecurity.class);
```

Comment:
Can't we directly use the extension here rather than assigning it to a object and then passing it to objectmapper